### PR TITLE
Store Deterministic forecasts with floats

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/hdf5_time_series_storage.jl [hunk 1]
```diff
@@ -126,8 +126,10 @@ function get_data_type(ts::TimeSeriesData)
         return "POLYNOMIAL"
     elseif data_type == PWL
         return "PWL"
+    elseif data_type <: Integer
```
Comment (daniel-thom) [inline src/hdf5_time_series_storage.jl:129]: We still have a problem converting SingleTimeSeries with integers to floats.  This is a workaround.

File: test/test_time_series.jl [hunk 2]
```diff
@@ -82,21 +78,15 @@ end
     other_time = initial_time + resolution
     name = "test"
     horizon = 24
-    data_vec = Dict(initial_time => ones(horizon), other_time => ones(horizon))
-    data_tuple =
-        Dict(initial_time => tuple(ones(horizon)...), other_time => tuple(ones(horizon)...))
```
Comment (daniel-thom) [inline test/test_time_series.jl:87]: I don't think we need to support this.  @sourabhdalvi @jd-lara Please comment if you disagree.

File: src/hdf5_time_series_storage.jl (hunk 1 above)
Comment (jd-lara) [reply src/hdf5_time_series_storage.jl:129]: Doesn't this happen automatically?

File: test/Project.toml [hunk 3]
```diff
@@ -11,6 +11,7 @@ Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
 Mustache = "ffc61752-8dc7-55ee-8c37-f3e9cdd09e70"
 PrettyTables = "08abe8d2-0d0c-5749-adfa-8a2ac140af0d"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
```
Comment (jd-lara) [inline test/Project.toml:14]: where do we use this?

File: test/test_time_series.jl (hunk 2 above)
Comment (jd-lara) [reply test/test_time_series.jl:87]: Not anymore, I added it to test the case where we convert to `Vector{Float64}`, given these changes this needs to be eliminated.

**Review by jd-lara — CHANGES_REQUESTED** [review]
The changes look reasonable to me. Needs to eliminate Tables from the toml file

File: src/hdf5_time_series_storage.jl (hunk 1 above)
Comment (daniel-thom) [reply src/hdf5_time_series_storage.jl:129]: Nope.  CONSTANT is now only Float64

File: test/Project.toml (hunk 3 above)
Comment (daniel-thom) [reply test/Project.toml]: Accident.  Thanks.

File: src/deterministic.jl [hunk 4]
```diff
@@ -205,9 +143,23 @@ function Deterministic(forecast::Deterministic, data::SortedDict{Dates.DateTime,
...
+    SortedDict{Dates.DateTime, Vector{CONSTANT}}(data...)
+convert_data(data::AbstractDict{Dates.DateTime, T}) where {T <: Vector{<:Tuple}} =
+    SortedDict{Dates.DateTime, Vector{POLYNOMIAL}}(data...)
+convert_data(data::AbstractDict{Dates.DateTime, T}) where {T <: Vector{<:Vector{<:Tuple}}} =
+    SortedDict{Dates.DateTime, Vector{PWL}}(data...)
+
+# Workaround for a bug/limitation in SortedDict. If a user tries to construct
+# SortedDict(i => ones(2) for i in 1:2)
+# it won't discern the types and will return SortedDict{Any,Any,Base.Order.ForwardOrdering}
+# https://github.com/JuliaCollections/DataStructures.jl/issues/239
```
Comment (daniel-thom) [inline src/deterministic.jl:156]: @jd-lara FYI, you can read the comments in this issue.

File: src/generated/Probabilistic.jl [hunk 5]
```diff
@@ -6,7 +6,7 @@ This file is auto-generated. Do not edit.
         name::String
         resolution::Dates.Period
         percentiles::Vector{Float64}
-        data::SortedDict{Dates.DateTime, Array}
+        data::Union{SortedDict{Dates.DateTime, Matrix{CONSTANT}}, SortedDict{Dates.DateTime, Matrix{POLYNOMIAL}}, SortedDict{Dates.DateTime, Matrix{PWL}}}
```
Comment (daniel-thom) [inline src/generated/Probabilistic.jl:9]: @jd-lara I made comparable changes for Probabilistic.  I suggest that we leave this PR open until I get a chance to fix Scenarios as well.

File: src/deterministic.jl (hunk 4 above)
Comment (jd-lara) [reply src/deterministic.jl:156]: Will check

File: src/hdf5_time_series_storage.jl (hunk 1 above)
Comment (jd-lara) [reply src/hdf5_time_series_storage.jl:129]: We should add a comment on why this is here then.

File: src/generated/Probabilistic.jl (hunk 5 above)
Comment (jd-lara) [reply src/generated/Probabilistic.jl:9]: That's ok, we should target wrapping up Monday though to focus on the pending items in PSY.

Comment (daniel-thom) [reply src/generated/Probabilistic.jl:9]: Also, I suspect that deserializing Probabilistic and Scenarios are broken with InMemoryTimeSeriesStorage.  There are no tests for it.  I'll address this.

Comment (daniel-thom) [reply src/generated/Probabilistic.jl:9]: @jd-lara I made comparable changes for Scenarios.  I did not make any changes for InMemoryTimeSeriesStorage.  It seems that you did not implement the required deserialize_time_series methods for that struct when you added Probabilistic and Scenarios.  I'll leave this to you.

**Review by jd-lara — APPROVED** [review]
LGTM thanks for the fix, this will make the interfaces more consistent

**Pull request merged**